### PR TITLE
Updated MOD_STAFF, SKIP and user object

### PR DESCRIPTION
**Pull request by Burkes**
I'd recommend also updating the Wiki for it, it says it returns the ID of who was skipped but even plug doesn't return that.

### CODE REVIEW

File: mapper.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ var serializeMedia = function(data) {
     } else {
         var title = utils.splitTitle(data.title);
 
-        media.author = title[0];
+        media.author = title[0];map
```
Comment (SooYou) [inline mapper.js]: typo

File: plugged.js [hunk 2]
```diff
@@ -654,19 +654,22 @@ Plugged.prototype._wsaprocessor = function(msg, flags) {
             break;
 
         case this.MOD_STAFF:
-            var promotion = mapper.mapPromotion(data.p);
+            var promotions = mapper.mapPromotion(data.p);
```
Comment (SooYou) [inline plugged.js]: the promotion array is always of length 1 except in one case and that is when the host passes his role over to another user. To my knowledge, there is no function that allows you to change the permissions of an array of users. You are fully right that this was missing, although the path taken is Ο(n²) and thus pretty slow. Instead I recommend to do a simple if clause for the host.

**Comment by SooYou** [comment]
Some good catches in there, I added some comments to clarify some parts. Also I'd like to mention that it was the case that plug used to chance the API every once in a while without an announcement, (certainly there is no information whatsoever in how far they will do this now since the owner has changed) which is part of why I really appreciate every contribution made.

It'd be a nice plus if you could clean up the commit messages a little and apply the style of the commits that has been made to the project so far :smile:

Thank you for your contribution to the project!

Have a nice day

**Comment by Burkes** [comment]
Oh, interesting, I made those changes on my own to later someone suggest me to do a pull request, and I'm not used to doing this. Sorry if most of the things are sloppy (mainly commit messages). I'll fix what you pointed out, but the promotion event wasn't updating the user rank in `state.room.users`. The change was mostly that, just happened to do the plural to make it more clear to whoever sees it.

If there is still anything missing, let me know. Again, I'm not good with GitHub :stuck_out_tongue:

**Comment by SooYou** [comment]
Don't worry, we're all constantly learning after all :stuck_out_tongue: 
The problem why the rank didn't update is that I used it as an object instead of as an array while the mapper returns the array. Changing it into a plural form was absolutely right (and the mapper method could be named to plural as well), what I meant though is the nested loop you added. Since the second user in the array can only be the host, I suggest (as in the comment before) to unwind the loop, which makes the function linear again, so there's a lot of time saving with little verbosity :smile:

**Comment by Burkes** [comment]
Oh well, sorry, I made my contributions but sincerely I think the change you mentioned would be a nanosecond difference, I'm not gonna bother doing it, if you want to take what I contributed, feel free to do so and edit it.

Besides I'm not sure how you'd look for the host without looping the array anyway.

**Comment by SooYou** [comment]
we're talking JS here haha. It'll approximately be around 70% increase in time +- 10% on V8, granted it happens rarely that the host is changed but it's time taken that could be used for other things. But fair enough :smile:, I'll apply the changes.

Have a nice weekend!

**Pull request merged**